src/plugins/intel_cpu/src/nodes/executors/fullyconnected_config.hpp

```diff
@@ -26,6 +26,9 @@ struct FCAttrs {
     MemoryCPtr decompressionSubtractPtr;
     MemoryCPtr decompressionMultiplyPtr;
     uint64_t dynamicQuantizationGroupSize;
+
+    // workload hint for FC to choose best kernels
+    bool isLLM = false;
```

Comment:
I would propose to extend https://github.com/openvinotoolkit/openvino/blob/e9d42dd60b961af2c2d92abde0a65ace6696542b/src/plugins/intel_cpu/src/nodes/executors/executor.hpp#L78 with modelType and use it directly im dnnl_fullyconnected.

Reply:
`useWeightsDecompressionImpl` needs modelType hint to determine if fp16 weights can be used, there are 3 call-sites:
 1. `noWeightsDecompression(const FCConfig& config)`
 2. `DnnlFCPrimitive::createShapeAgnosticData(const FCAttrs& attrs, , ,const ExecutorContext::CPtr context,)`
 3. `DnnlFCPrimitive::DnnlFCPrimitive(const Key& key, const dnnl::engine& engine, const std::vector<impl_desc_type>& implPriorities)`

`ExecutorContext`  is only available on site 2, but `FCAttrs` is available on call-sites 1 & 2, so I prefer to pass modelType through `FCAttrs` (to reduce damage to the code structure).

On call-site 3, I added modelType into the key.